flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertexProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.executiongraph;
+
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+
+/**
+ * {@code ExecutionVertexProvider} offers methods for retrieving {@link AccessExecutionVertex}
+ * instances.
+ *
+ * @param <T> The type of {@code AccessExecutionVertex}.
+ */
+public interface ExecutionVertexProvider<T extends AccessExecutionVertex> {
+
+    /**
+     * Returns all {@link ExecutionVertex} instances that are handled by this provider.
+     *
+     * @return All {@code ExecutionVertex} instances.
+     */
+    Iterable<T> getAllExecutionVertices();
+
+    /**
+     * Returns the {@link ExecutionVertex} for the given {@link ExecutionVertexID}.
+     *
+     * @param executionVertexId The ID of the {@code ExecutionVertex}.
+     * @return The {@code ExecutionVertex}.
+     * @throws IllegalArgumentException if the passed {@code ExecutionVertexID} is not associated to
+     *     a {@code ExecutionVertex} handled by this provider.
+     */
+    T getExecutionVertex(ExecutionVertexID executionVertexId);
```

Comment:
This interface seems weird given that the history service only cares about executions.